Common/src/DependencyResolver/DependencyResolver.cs

```diff
@@ -0,0 +1,256 @@
+  /// <param name="pullQueueStorage">Interface to get tasks from the queue</param>
+  /// <param name="pushQueueStorage">Interface to put tasks in the queue</param>
+  /// <param name="taskTable">Interface to manage task states</param>
+  /// <param name="resultTable">Interface to manage result states</param>
+  /// <param name="logger">Logger used to produce logs for this class</param>
+  public DependencyResolver(IPullQueueStorage           pullQueueStorage,
+                            IPushQueueStorage           pushQueueStorage,
+                            ITaskTable                  taskTable,
+                            IResultTable                resultTable,
+                            ILogger<DependencyResolver> logger)
+  {
+    pullQueueStorage_ = pullQueueStorage;
+    pushQueueStorage_ = pushQueueStorage;
+    taskTable_        = taskTable;
+    resultTable_      = resultTable;
+    logger_           = logger;
+  }
+
+  /// <inheritdoc />
+  public async Task<HealthCheckResult> Check(HealthCheckTag tag)
+  {
+    var checks = new List<Task<HealthCheckResult>>
+                 {
+                   pullQueueStorage_.Check(tag),
+                   pushQueueStorage_.Check(tag),
+                   resultTable_.Check(tag),
+                   taskTable_.Check(tag),
+                 };
+
+    var exceptions  = new List<Exception>();
+    var data        = new Dictionary<string, object>();
+    var description = new StringBuilder();
+    var worstStatus = HealthStatus.Healthy;
+
+    foreach (var healthCheckResult in await checks.WhenAll()
+                                                  .ConfigureAwait(false))
+    {
+      if (healthCheckResult.Status == HealthStatus.Healthy)
+      {
+        continue;
+      }
+
+      if (healthCheckResult.Exception is not null)
+      {
+        exceptions.Add(healthCheckResult.Exception);
+      }
+
+      foreach (var (key, value) in healthCheckResult.Data)
+      {
+        data[key] = value;
+      }
+
+      if (healthCheckResult.Description is not null)
+      {
+        description.AppendLine(healthCheckResult.Description);
+      }
+
+      worstStatus = worstStatus < healthCheckResult.Status
+                      ? worstStatus
+                      : healthCheckResult.Status;
+    }
+
+    return new HealthCheckResult(worstStatus,
+                                 description.ToString(),
+                                 new AggregateException(exceptions),
+                                 data);
+  }
+
+  /// <inheritdoc />
+  public async Task Init(CancellationToken cancellationToken)
+  {
+    await pushQueueStorage_.Init(cancellationToken)
+                           .ConfigureAwait(false);
+    await pullQueueStorage_.Init(cancellationToken)
+                           .ConfigureAwait(false);
+    await resultTable_.Init(cancellationToken)
+                      .ConfigureAwait(false);
+    await taskTable_.Init(cancellationToken)
+                    .ConfigureAwait(false);
```

Comment:
Would it make sense to make them in parallel?
```suggestion
    await Task.WhenAll(new []
                       {
                         pushQueueStorage_.Init(cancellationToken),
                         pullQueueStorage_.Init(cancellationToken),
                         resultTable_.Init(cancellationToken),
                         taskTable_.Init(cancellationToken),
                       })
              .ConfigureAwait(false);
```

Reply:
Well, it was not done in parallel in the other Init so I did the same. It makes sense but I'm not sure if it will break something.